rest/view_api_test.go

```diff
@@ -277,7 +304,12 @@ func TestUserViewQuery(t *testing.T) {
 // This includes a fix for #857
 func TestAdminReduceViewQuery(t *testing.T) {
 
-	rt := RestTester{SyncFn: `function(doc) {channel(doc.channel)}`}
+	// Since this test uses views (and assumes no GSI indexes), then must explicitly tell the RestTester
+	// to use views, since otherwise will default to GSI
+	dbConfig := &DbConfig{
+		UseViews: true,
+	}
+	rt := RestTester{SyncFn: `function(doc) {channel(doc.channel)}`, DatabaseConfig: dbConfig}
```

Comment:
This is creating a custom (non-SG) view definition, so it's unrelated whether the RestTester is using GSI for internal queries.

If the motivation is just to avoid GSI overhead for tests that don't require the built-in SG queries, I think we should come up with a more generic way of doing that.

Reply:
The motivation for this was basically in reaction to issues I hit trying to get a clean run:

https://github.com/couchbase/sync_gateway/issues/3570#issuecomment-390487982

Without `UseViews: true`, `InitializeViews()` (which waits for views to be ready) is not being called, and I was seeing `view_undefined` errors.  (in my original comment I said `InitializeIndexes()`, but I must've meant `InitializeViews()`)

I guess what you're saying is that this change will cause it to no longer exercise the underlying GSI queries?   I guess for these tests, we want to call `InitializeIndexes()` **and** `InitializeViews()`?

Reply:
The views being queried in tests like TestViewQueryMultipleViews (mentioned on the linked comment) aren't initialized during InitializeViews - they are created by the test.  I'd like to understand the specific point of failure for these tests, and why they aren't failing on an integration run against master, that would also not be calling InitializeViews().

Reply:
It appears to be a race between the view creation API call and the view being ready for query.  This effectively works around it:

```
func (rt *RestTester) WaitForNViewResults(numResultsExpected int, viewUrlPath string, user auth.User, password string) (viewResult sgbucket.ViewResult, err error) {

	worker := func() (shouldRetry bool, err error, value interface{}) {
		...
		if strings.Contains(response.Body.String(), "view_undefined") {
			base.Infof(base.KeyAll, "view_undefined error: %v.  Retrying", response.Body.String())
			return true, nil, nil
		}

```

logs: https://gist.github.com/tleyden/9c36f0a55120d3076beb8c03f8099f85#file-gistfile1-txt-L28

Reply:
Fixed in 9125d1e5ddfa7c9b1b5be0c2dadf32390e4f265c using the above retry workaround.  Reverted `DbConfig{ UseViews: true } changes` in  938cbf6d0e51c364abdb56df1d124c3fe6d233bc.

Kicking off integration test re-runs